src/o365/teams/commands/deeplink/tab-generate.ts

```diff
@@ -0,0 +1,188 @@
+    return 'Generates a Microsoft Teams deep link from an existing Tab in a Channel';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/teams/${encodeURIComponent(args.options.teamId)}/channels/${encodeURIComponent(args.options.channelId)}/tabs/${encodeURIComponent(args.options.tabId)}?$expand=teamsApp`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      json: true
+    }
+
+    request
+      .get<Tab>(requestOptions)
+      .then((res: Tab): void => {
+        let appId: string = res.teamsApp.id;
+        let entityId: string = res.configuration.entityId;
+        let contentUrl: string = encodeURIComponent(res.webUrl);
+        let deeplink: Deeplink = { deeplink: "" };
+
+        let tabTypeInput: string = args.options.tabType ? args.options.tabType.trim() : TabTypeOptions.Static;
+
+        if (TabTypeOptions[(tabTypeInput as keyof typeof TabTypeOptions)].valueOf() == TabTypeOptions.Configurable) {         
+            let context: string = `{"channelId": "${args.options.channelId}"}`;
+            deeplink = { deeplink: `https://teams.microsoft.com/l/entity/${appId}/${entityId}?webUrl=${contentUrl}&label=${args.options.label}&context=${context}` };
+            cmd.log(deeplink);
+        }
+        else if (TabTypeOptions[(tabTypeInput as keyof typeof TabTypeOptions)].valueOf() == TabTypeOptions.Static) {
+            deeplink = { deeplink: `https://teams.microsoft.com/l/entity/${appId}/${entityId}?webUrl=${contentUrl}&label=${args.options.label}` };
+            cmd.log(deeplink);
+        }
+
+        if (this.verbose) {
+          cmd.log(vorpal.chalk.green('DONE'));
+        }
+
+        cb();
+      }, (err: any) => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  /**
+   * Maps the base TabTypeOptions enum to string array so it can 
+   * more easily be used in validation or descriptions.
+   */
+  protected get tabTypeMap(): string[] {
+    const result: string[] = [];
+
+    for (let tabType in TabTypeOptions) {
+      result.push(tabType);
+    }
+    return result;
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-i, --teamId <teamId>',
+        description: 'The ID of the team where the tab exists'
+      },
+      {
+        option: '-c, --channelId <channelId>',
+        description: 'The ID of the channel where the tab exists'
+      },
+      {
+        option: '-t, --tabId <tabId>',
+        description: 'The ID of the tab to generate the deep link from'
+      },
+      {
+        option: '-l, --label <label>',
+        description: 'The label to use in the deep link'
+      },
+      {
+        option: '-m, --tabType <TabTypeOptions>',
+        description: `The tab type. Allowed values ${this.tabTypeMap.join('|')}. Default ${this.tabTypeMap[0]}`,
+        autocomplete: this.tabTypeMap
```

Comment:
As we only have two possible values for this option and they are not likely to change, we can just use a string array here to keep things simple and easier to read, as we have done with other commands.

See example here in `cdn-policy-set.ts`

https://github.com/pnp/office365-cli/blob/9b70527cc776495e73e5644b2b271cb120a26871/src/o365/spo/commands/cdn/cdn-policy-set.ts#L106-L110